modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/TaskActionExecutor.java

```diff
@@ -231,19 +239,47 @@ private boolean callActionServiceMethod(ActionEvent action, Map<String, Object>
             try {
                 Method method = service.getClass().getMethod(serviceMethod, methodHandler.getClasses());
 
-                try {
-                    method.invoke(service, methodHandler.getObjects());
-                } catch (IllegalAccessException | InvocationTargetException e) {
-                    throw new TaskHandlerException(
-                            ACTION, "task.error.serviceMethodInvokeError", e,
-                            serviceMethod, action.getServiceInterface()
-                    );
+                int numberOfRetries = task.getNumberOfRetries();
```

Comment:
All of this reeks of overcomplication. If we are extending the event anyway, why not just put the max retry number inside the event?